gcpvault.go

```diff
@@ -52,6 +52,10 @@ type Config struct {
 	// MetadataAddress is the location of the GCP metadata
 	// This should only used for testing.
 	MetadataAddress string `envconfig:"METADATA_ADDR"`
+
+	// HTTPClient allows for providing a custom *http.Client. If not provided, one
+	// will be chosen based on system architecture.
+	HTTPClient *http.Client
```

Comment:
Sorry, I'm a bit confused...it doesn't look like this client is used anywhere?

Also, in which scenario does the internal `getHTTPClient(ctx)` func not work? Is there some point where we don't have a `context.Context` available for `urlfetch` to do it's thing?

Reply:
when running unit tests it defaults to the gcp httpClient, not the appengine one. I've modified `newClient` below as such

```
	vcfg.HttpClient = cfg.HTTPClient
	if vcfg.HttpClient == nil {
		vcfg.HttpClient = getHTTPClient(ctx)
	}```